agents/ibootbar/ibootbar.py

```diff
@@ -0,0 +1,520 @@
+                self.ibootbar.names = []
+
+                # Send get command and append results to self.ibootbar.names
+                result = yield self.ibootbar.snmp.get(get_list, self.ibootbar.version)
+                for i in range(len(result)):
+                    self.ibootbar.names.append(result[i][1].prettyPrint())
+
+                return True, "The outlet names are: {}".\
+                    format(self.ibootbar.names)
+            else:
+                return False, "Could not acquire lock"
+
+    @ocs_agent.param('outlet', choices=[0, 1, 2, 3, 4, 5, 6, 7])
+    @ocs_agent.param('state', choices=['on', 'off'])
+    @inlineCallbacks
+    def set_outlet(self, session, params=None):
+        """
+        **Task** - Sets a particular outlet to on/off
+        """
+        with self.lock.acquire_timeout(3, job='set_outlet') as acquired:
+            if acquired:
+                # Convert given state parameter to integer
+                if params['state'] == 'on':
+                    state = 1
+                else:
+                    state = 0
+
+                # Send set command to given outlet
+                outlet = [('IBOOTPDU-MIB', 'outletControl', params['outlet'])]
+                setcmd = yield self.ibootbar.snmp.set(outlet, self.ibootbar.version, state)
+                print(setcmd)
+
+                # Force get status request by rewinding the lastGet time by 60 seconds
+                for mib in self.ibootbar.mib_timings:
+                    mib["lastGet"] = mib["lastGet"] - 60
+
+                return True, 'Set outlet {} to {}'.\
+                    format(params['outlet'], params['state'])
+            else:
+                return False, "Could not acquire lock"
+
+    @ocs_agent.param('outlet', choices=[0, 1, 2, 3, 4, 5, 6, 7])
+    @ocs_agent.param('cycle_time', default=10, type=int)
+    @inlineCallbacks
+    def cycle_outlet(self, session, params=None):
+        """
+        **Task** - Cycles a particular outlet for given amount of seconds
+        """
+        with self.lock.acquire_timeout(3, job='cycle_outlet') as acquired:
+            if acquired:
+                # Send set command for cycle time
+                set_cycle = [('IBOOTPDU-MIB', 'outletCycleTime', params['outlet'])]
+                setcmd1 = yield self.ibootbar.snmp.set(set_cycle, self.ibootbar.version, params['cycle_time'])
+                print(setcmd1)
+
+                # Send set command to given outlet
+                outlet = [('IBOOTPDU-MIB', 'outletControl', params['outlet'])]
+                setcmd2 = yield self.ibootbar.snmp.set(outlet, self.ibootbar.version, 2)
+                print(setcmd2)
+
+                return True, 'Cycling outlet {} for {} seconds'.\
+                    format(params['outlet'], params['cycle_time'])
+            else:
+                return False, "Could not acquire lock"
+
+    @ocs_agent.param('_')
+    @inlineCallbacks
+    def reboot(self, session, params=None):
```

Comment:
What happens to the SNMP communication going on in `acq` during a reboot? Does it lose communication, or gracefully reconnect when the ibootbar is back online? Related, what state does the acq process show for all outlets during that time? (It should show they're all off.)